twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
```

Comment:
Maybe it's possible to implement the `_MakeDeferredAwaitable` class as a generator... The performance would be a bit better (although the code would be harder to read...)